src/main/python/deep_qa/training/text_trainer.py

```diff
@@ -360,34 +359,36 @@ def _get_new_embedding(self, name: str, embedding_size: int, vocab_name: str='wo
                                                name=name + '_projection')
         return embedding_layer, projection_layer
 
-    def _get_sentence_encoder(self):
+    def _get_sentence_encoder(self, encoder_type="default"):
         """
         A sentence encoder takes as input a sequence of word embeddings, and returns as output a
         single vector encoding the sentence.  This is typically either a simple RNN or an LSTM, but
         could be more complex, if the "sentence" is actually a logical form.
         """
-        if self.sentence_encoder_layer is None:
-            self.sentence_encoder_layer = self._get_new_sentence_encoder()
-        return self.sentence_encoder_layer
+        if encoder_type not in self.sentence_encoder_layers:
+            encoder_layer_name = encoder_type + "_sentence_encoder"
+            new_sentence_encoder = self._get_new_sentence_encoder(encoder_type, encoder_layer_name)
```

Comment:
You can remove `_get_new_sentence_encoder` method, instead just using `_get_new_encoder`.